app/bundles/LeadBundle/Entity/LeadList.php

```diff
@@ -306,8 +306,8 @@ public function getFilters()
 
     public function needsRebuild(): bool
     {
-        // Manual segments never require rebuild
-        if (empty($this->getFilters())) {
+        // Manual or unpublished segments never require rebuild
+        if (empty($this->getFilters() || !$this->isPublished())) {
```

Comment:
```suggestion
        if (empty($this->getFilters()) || !$this->isPublished()) {
```

Brackets look to be in the wrong place.

This should be possible to add a test for as well can you add a test?

I had this on my backlog but didn't get time to look at it yet - thanks for looking at it!

Reply:
For the test this is extra work not needed, we fix an issue and don't reduce the code coverage. We'll stay with this proposition. Thank you a lot for your quick review.